tensorflow/core/kernels/data/name_utils.cc

```diff
@@ -45,6 +45,14 @@ string OpName(const string& dataset_type) {
   }
 }
 
+string OpName(const string& dataset_type, int op_version) {
```

Comment:
It seems odd to have a version of the method that accepts the op version and have it call a method that then uses regexp to see if there is an op version.

We should either have one op that operates over an opaque string and uses a regexp. Or two version in which case the version-less method ends up calling the one with the version argument.

Reply:
Yes, the current implementation is odd. The new API is proposed in the below comment.